frontends/web/src/components/forms/input.tsx

```diff
@@ -15,39 +15,20 @@
  * limitations under the License.
  */
 
-import { forwardRef } from 'react';
+import { ChangeEvent, HTMLProps, forwardRef } from 'react';
 import styles from './input.module.css';
 
-type withValue = {
-  value: string | number;
-  defaultValue?: string | number;
-} | {
-  defaultValue: string | number;
-}
 
-export type Props = withValue & {
+
+export type Props = {
     align?: 'left' | 'right';
-    autoFocus?: boolean;
     children?: React.ReactNode;
     className?: string;
-    disabled?: boolean;
     error?: string | object;
-    id?: string;
-    label?: string;
-    min?: string;
-    name?: string;
-    onInput?: (e: any) => void;
-    onPaste?: (e: any) => void;
-    pattern?: string;
-    placeholder?: string;
-    readOnly?: boolean;
-    step?: string;
-    title?: string;
+    onInput?: (e: ChangeEvent<HTMLInputElement>) => void;
     transparent?: boolean;
-    type?: 'text' | 'password' | 'number';
-    maxLength?: number;
     labelSection?: JSX.Element | undefined;
-}
+} & Omit<HTMLProps<HTMLInputElement>, 'onInput'>
```

Comment:
> however I'm unsure if there's a specific reason that made us to use onInput instead of onChange

afaik onInput triggers on every change, and onChange fires when the element looses focus. So we prefer onInput for validation etc.

Reply:
oh good to know.. in this case I think our Input component should have `& Omit<HTMLProps<HTMLInputElement>, 'onInput'>` so that we can use onInput and onChange...


found a link to an explanation in your SO link https://github.com/facebook/react/issues/9567#issuecomment-298326902